internal/catalogmetadata/cache/cache.go

```diff
@@ -95,7 +95,11 @@ func (fsc *filesystemCache) FetchCatalogContents(ctx context.Context, catalog *c
 		return nil, fmt.Errorf("error forming request: %v", err)
 	}
 
-	resp, err := fsc.client.Do(req)
+	client, err := fsc.getClient()
+	if err != nil {
+		return nil, fmt.Errorf("error getting HTTP client: %w", err)
+	}
```

Comment:
As far as I understand this PR - we are creating a new client each time we need to make a request to catalogd to make sure that we read a new cert.

But if it is the case - why do we need a cert watcher?

Reply:
Because we don't want to read the certs off disk every single time we want to make a connection. The cert pool is used in multiple places.

Reply:
Makes sense. Resolving this question.